src/erlcloud_aws.erl

```diff
@@ -363,6 +370,39 @@ config_env() ->
         _ -> {error, environment_config_unavailable}
     end.
 
+config_with_region(Config) ->
+    case {os:getenv("AWS_REGION"), application:get_env(erlcloud, aws_region)} of
+        {false, undefined} -> Config;
+        {false, {ok, Region}} -> config_with_region(Region, Config);
+        {Region, _} -> config_with_region(Region, Config)
+    end.
+
+config_with_region(Region, Config) ->
+    Services =
+        [autoscaling,
+         cloudformation,
+         cloudsearch,
+         cloudtrail,
+         dynamodb,
+         'streams.dynamodb',
+         ec2,
```

Comment:
yes, this list is bad. Perhaps code that traverses the #aws_config{} record and looks for tags of the form *_host would be more future proof?